api/src/declare/declare.controller.ts

```diff
@@ -0,0 +1,20 @@
+import { Controller, Get, HttpException, HttpStatus } from '@nestjs/common';
+import { DeclareService } from './declare.service';
+
+@Controller('declare')
+export class DeclareController {
+  constructor(private readonly declareService: DeclareService) {}
+
+  @Get('list')
+  async listDeclarations() {
+    try {
+      return await this.declareService.getDeclarations();
+    } catch (error) {
+      console.error('Error at declare list:', error);
+      throw new HttpException(
+        { message: 'Error fetching declarations', error },
+        HttpStatus.BAD_REQUEST,
```

Comment:
If an error is being thrown out of the service, then the error would be an internal error, not a 400 Bad request, no?